### ISSUE TITLE
middleName not present in getCustomerName from Order

### ISSUE THREAD

**Issue by joinso** [magento/magento2#23627]
### Preconditions (*)

In this scenario:
1. Magento 2.3

### Steps to reproduce (*)

1. In Magento Admin, setup MiddleName as visible on Customer Configuration
2. Register a new customer and provide First Name, Middle Name and Last Name.
3.- Place an order.

### Expected result (*)

1. In Email of new order, you must see "First Name" "Middle Name" "Last Name"
2. Also in backend, you must see "First Name" "Middle Name" "Last Name" in customer details on order.

### Actual result (*)

1. Middle Name is missing: only "First Name" "Last Name" is shown.

The bug is in this class: vendor/magento/module-sales/Model/Order.php
In method getCustomerName():

    public function getCustomerName()
    {
        if ($this->getCustomerFirstname()) {
$customerName = $this->getCustomerFirstname() . ' ' . $this->getCustomerLastname();

        } else {
            $customerName = (string)__('Guest');
        }
        return $customerName;
    }
the middleName is not appended.

So, line:

$customerName = $this->getCustomerFirstname() . ' ' . $this->getCustomerLastname();

must be replaced with:

$customerName = $this->getCustomerFirstname() . '  ' . $this->getCustomerMiddlename() . ' ' . $this->getCustomerLastname();

Or with another one that take care of the configuration of the Middle Name.

Regards,
JOINSO

PD: Perhaps someone can submit a patch to avoid the problem until this is solved in a next release.

**Comment by krishprakash** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @krishprakash. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @krishprakash, here is your [Magento instance](https://i-23627-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23627-2-3-develop.instances.magento-community.engineering/admin](https://i-23627-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @krishprakash 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-17990`, `MC-17991` were created

**Issue Available**: @krishprakash, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @krishprakash 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-17990`, `MC-17991` were created

**Issue Available**: @krishprakash, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by gwharton** [comment]
Shouldn't any proposed solution respect the Magento Settings 

```
Stores -> Configuration -> Customers -> Customer Configuration -> Name and Address Options -> Show Prefix
Stores -> Configuration -> Customers -> Customer Configuration -> Name and Address Options -> Show Middle Name
Stores -> Configuration -> Customers -> Customer Configuration -> Name and Address Options -> Show Suffix
```

**Comment by Bartlomiejsz** [comment]
Hi @magento-engcom-team, shouldn't this be closed since #24746 is merged?

**Comment by raumatbel** [comment]
Now is merged in the branch 2.3-develop.

**Comment by plastikschnitzer** [comment]
Is this fixed in Magento 2.3.5-p1? For me its not working there, middle name is showing up in the customer administration only, not in the order overview, order detail page or the transactional email.
